version: '3.8'

services:
  db:
    image: postgres:12.4
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_DB: ${POSTGRES_DB}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
    volumes:
      - todolist_pg_data:/var/lib/postgresql/data
    healthcheck:
      test: pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}
      timeout: 3s
      retries: 10
      interval: 3s

  bot:
    image: freddy111/todolist:latest
    env_file: .env
    depends_on:
      db:
        condition: service_healthy
    command: python manage.py runbot

  api:
    image: freddy111/todolist:latest
    env_file: .env
    depends_on:
      db:
        condition: service_healthy

  collect_static:
    image: freddy111/todolist:latest
    env_file: .env
    entrypoint: ''
    volumes:
      - django_static:/opt/static/
    command: python3 manage.py collectstatic -c --no-input

  frontend:
    image: sermalenk/skypro-front:lesson-38
    ports:
      - "80:80"
    depends_on:
      collect_static:
        condition: service_completed_successfully
    volumes:
      - django_static:/usr/share/nginx/html/static/
      - ./nginx.conf:/etc/nginx/conf.d/default.conf

volumes:
  todolist_pg_data:
  django_static:
